### ISSUE TITLE
Calling `code .` from WSL prompt does not return

### ISSUE THREAD

**Issue by sarunint** [Microsoft/vscode#62698]
- VSCode Version: 1.28.2
- OS Version: Windows 10 v1809

Steps to Reproduce:

1. Run `code .` from WSL
2. The WSL prompt does not return.
3. Type `exit` into WSL without closing VSCode, WSL window does not close.

Please note that this issue has been there all along since I started using VSCode. (2017, maybe?) Suddenly, it got fixed sometime in 2018. (I can't remember when, unfortunately.) And it has been reintroduce again last month. (When the UI for Windows version has been redesigned.)

Does this issue occur when all extensions are disabled?: Yes

**Comment by sarunint** [comment]
Here's the screenshot, for clarity:
![image](https://user-images.githubusercontent.com/3850197/48114694-a8ce6000-e292-11e8-8044-57edba42d2a7.png)

**Comment by sandy081** [comment]
@dbaeumer Forwarding it to you.

**Comment by dbaeumer** [comment]
Actually this support got added by @Tyriar. Assigning to him.

**Comment by Tyriar** [comment]
This doesn't reproduce for me on 1.29.1 and Windows 1809 (OS build 18282.1000). Both opening the first window and additional windows returns the prompt correctly for me.

**Comment by Tyriar** [comment]
I do see logout but no window close after I exit, but this looks like a problem for https://github.com/Microsoft/console

![image](https://user-images.githubusercontent.com/2193314/49259150-07e44800-f3ed-11e8-9370-64f5e5d4daa9.png)

**Comment by sarunint** [comment]
@Tyriar Hi, I'm on Windows 10 Version 1809 Build 17763.134 and VSCode 1
29.1. And the issue is present.

I'll try to reproduce it on other computers

**Comment by karlhorky** [comment]
Also seeing this problem when using [WSL (Ubuntu) over Hyper](https://medium.com/@ssharizal/hyper-js-oh-my-zsh-as-ubuntu-on-windows-wsl-terminal-8bf577cdbd97).

@Tyriar @sarunint which programs are you two opening to get to that terminal?

**Comment by Tyriar** [comment]
@karlhorky this is talking about the vscode CLI so it shouldn't have anything to do with hyper? The program conhost running wsl.exe

**Comment by Tyriar** [comment]
@karlhorky oh I misunderstood, you're running the code CLI via hyper, that sounds like this.

**Comment by karlhorky** [comment]
Yeah for me the reproduction steps are:

1. Start Windows 10
2. Open Hyper (with config: `shell: 'C:\\Windows\\System32\\bash.exe'` in `.hyper.js`)
3. Switch to project directory and type `code .`
4. Prompt hangs

At step 4, you can <kbd>Ctrl</kbd>-<kbd>C</kbd> to send a `SIGINT` and get back to a working prompt (VS Code will stay open), but it's kind of annoying.

**Comment by weikinhuang** [comment]
Might be related to my issue here? https://github.com/Microsoft/vscode/issues/63539

**Comment by karlhorky** [comment]
Sounds very similar @weikinhuang! Maybe your issue was also misinterpreted...

**Comment by BookOfGreg** [comment]
Worked correctly in 1803, does not work in 1809.
This is likely a duplicate of https://github.com/Microsoft/WSL/issues/3734

**Comment by robhawkes** [comment]
Also seeing this. Calling `code .` from within WSL (using Hyper) returned immediately prior to updating to 1809, however after updating the command doesn't return until I force it (using `ctrl + c`).

**Comment by ccpacillos** [comment]
I have a very similar issue. I have been following this thread and checking it every single day.
From what I have observed, if you have VSCode app already running, calling `code .` from wsl will return immediately.

**Comment by ccpacillos** [comment]
Following the issue of @weikinhuang, adding a function that calls `code` with `| cat` is my temporary solution for now. I just really hate doing `Ctrl+c` whenever I run `code`.

I wanted to retain the use of the `code` command instead of another function name so I did the following:
1. From wsl, I renamed `/usr/bin/code` to `/usr/bin/vscode`. This is a symlink that points to the vscode shell script launcher from Windows itself.
2. Added this function:
```bash
vcode() {
  vscode $1 | cat
}
```
3. Create an alias `alias code="vcode"`.

I think there's a better way of dealing with this and still use the same good old `code` command.
